### PR TITLE
Update dependencies and exports

### PR DESCRIPTION
**Pull request by wilwade**
A lot of dependencies were out of date. This brings all but the test framework ones to latest.

Also adds the export for declare to have more types exported at the root instead of having to dig.

### CODE REVIEW

File: parquet.ts [hunk 1]
```diff
@@ -4,6 +4,8 @@ import * as schema from './lib/schema';
 import * as shredder from './lib/shred';
 import * as fields from './lib/fields';
 
+export * from './lib/declare';
```
Comment (wilwade) [inline parquet.ts:7]: Export more types for easier use.

File: eslint.config.mjs [hunk 2]
```diff
@@ -11,10 +11,11 @@ export default tseslint.config(
   ...tseslint.configs.strict,
   ...tseslint.configs.stylistic,
   {
-    ignores: ['dist/*', 'test/browser/main.js'],
+    ignores: ['gen-nodejs/*', 'dist/**/*', 'dist/*', 'test/browser/main.js'],
   },
   {
     rules: {
+      'no-loss-of-precision': 'off',
```
Comment (wilwade) [inline eslint.config.mjs:18]: This is so we don't get double warnings. See line 20.

File: eslint.config.mjs [hunk 3]
```diff
@@ -11,10 +11,11 @@ export default tseslint.config(
   ...tseslint.configs.strict,
   ...tseslint.configs.stylistic,
   {
-    ignores: ['dist/*', 'test/browser/main.js'],
+    ignores: ['gen-nodejs/*', 'dist/**/*', 'dist/*', 'test/browser/main.js'],
```
Comment (wilwade) [inline eslint.config.mjs:14]: Merged with the other one that was below this

File: eslint.config.mjs [hunk 4]
```diff
@@ -44,6 +45,15 @@ export default tseslint.config(
     },
   },
   {
-    ignores: ['gen-nodejs/*', 'dist/**/*'],
+    files: ['**/*.test.ts', 'test/**/*', 'examples/**/*'],
```
Comment (wilwade) [inline eslint.config.mjs:48]: Ignore some things in tests.

**Review by enddynayn — APPROVED** [review]
great! 👍

**Pull request merged**